### PR TITLE
Refactoring questionnaire

### PR DESCRIPTION
**Pull request by benjefferies**
### What is the context of this PR?
* Removing unused code paths
* Merging identical logic
* Defining named parameters for clarity

### How to review 
All tests should pass, suggest further improvements or object any changes

### CODE REVIEW

File: app/questionnaire/location.py [hunk 1]
```diff
@@ -51,31 +51,25 @@ def url(self, metadata):
         collection_id = metadata["collection_exercise_sid"]
         form_type = metadata["form_type"]
 
+        path_params = {
```
Comment (jonnyshaw89) [inline app/questionnaire/location.py:54]: nice

File: app/questionnaire/questionnaire_manager.py [hunk 2]
```diff
@@ -73,24 +68,24 @@ def process_incoming_answers(self, location, post_data):
 
         return is_valid
 
-    def build_state(self, location, answers):
+    def build_block_state(self, location, answers):
```
Comment (jonnyshaw89) [inline app/questionnaire/questionnaire_manager.py:71]: hopefully this whole function should go with WTForms

**Review by jonnyshaw89 — APPROVED** [review]
nice tidy up. The naming of the parameters adds a lot more context

File: app/questionnaire/location.py [hunk 3]
```diff
@@ -51,31 +51,25 @@ def url(self, metadata):
...
-                       collection_id=collection_id,
-                       group_id=self.group_id,
-                       group_instance=self.group_instance,
-                       block_id=self.block_id)
+                return url_for('questionnaire.get_thank_you', **path_params)
+        elif self.block_id == 'confirmation':
+            return url_for('questionnaire.get_confirmation', **path_params)
+        else:
+            return url_for('questionnaire.get_block',
+                           eq_id=eq_id,
```
Comment (iwootten) [inline app/questionnaire/location.py]: This can use path params as well can't it?

File: app/questionnaire/questionnaire_manager.py (hunk 2 above)
Comment (iwootten) [reply app/questionnaire/questionnaire_manager.py:71]: Whole class :)

File: tests/integration/questionnaire/test_questionnaire_final_confirmation.py [hunk 4]
```diff
@@ -36,7 +36,6 @@ def test_final_confirmation_asked_at_end_of_questionnaire(self):
         resp = self.client.get(resp.headers['Location'], follow_redirects=False)
         self.assertEquals(resp.status_code, 200)
 
-
```
Comment (iwootten) [inline tests/integration/questionnaire/test_questionnaire_final_confirmation.py:39]: This is the best change of the whole PR

**Review by iwootten — CHANGES_REQUESTED** [review]

File: app/questionnaire/location.py (hunk 3 above)
Comment (benjefferies) [reply app/questionnaire/location.py]: Done

File: app/questionnaire/questionnaire_manager.py (hunk 2 above)
Comment (benjefferies) [reply app/questionnaire/questionnaire_manager.py:71]: Yeah at some point I hope we can remove it with WTForm but trying to make it maintainable in the short term

File: tests/integration/questionnaire/test_questionnaire_final_confirmation.py (hunk 4 above)
Comment (benjefferies) [reply tests/integration/questionnaire/test_questionnaire_final_confirmation.py:39]: Thanks for the acknowledgement 👍

**Review by iwootten — APPROVED** [review]

**Pull request merged**